### PR TITLE
Set balance refactor

### PR DESCRIPTION
**Pull request by sasurobert**

### CODE REVIEW

File: integrationTests/vm/iele/vmAgar_test.go [hunk 1]
```diff
@@ -193,6 +194,8 @@ func TestAgarioContractJoinGameReward(t *testing.T) {
 		t.Skip("this is not a short test")
 	}
 
+	_ = logger.SetLogLevel("*TRACE")
```
Comment (iulianpascalau) [inline integrationTests/vm/iele/vmAgar_test.go]: maybe remove this

File: integrationTests/vm/iele/vmAgar_test.go [hunk 2]
```diff
@@ -222,7 +225,7 @@ func TestAgarioContractJoinGameReward(t *testing.T) {
 	defaultUserNonce := uint64(10)
 	defaultUserBalance := big.NewInt(100000000)
 
-	noOfUsers := 10
+	noOfUsers := 1
```
Comment (iulianpascalau) [inline integrationTests/vm/iele/vmAgar_test.go]: why this change?

File: process/rewardTransaction/process.go [hunk 3]
```diff
@@ -85,10 +83,7 @@ func (rtp *rewardTxProcessor) ProcessRewardTransaction(rTx *rewardTx.RewardTx) e
 
 	process.DisplayProcessTxDetails("ProcessRewardTransaction: receiver account details", accHandler, rTx)
 
-	operation := big.NewInt(0)
-	operation = operation.Add(rTx.Value, rewardAcc.Balance)
-	err = rewardAcc.SetBalanceWithJournal(operation)
-
+	err = rewardAcc.AddToBalance(rTx.Value)
```
Comment (iulianpascalau) [inline process/rewardTransaction/process.go]: :+1: nice and clean

File: integrationTests/vm/iele/vmAgar_test.go (hunk 1 above)
Comment (sasurobert) [reply integrationTests/vm/iele/vmAgar_test.go]: reverted

File: integrationTests/vm/iele/vmAgar_test.go (hunk 2 above)
Comment (sasurobert) [reply integrationTests/vm/iele/vmAgar_test.go]: reverted

File: data/state/account.go [hunk 4]
```diff
@@ -225,11 +227,26 @@ func (a *Account) AddToDeveloperReward(value *big.Int) error {
...
+
+	err := a.setBalanceWithJournal(newBalance)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// SubFromBalance adds new value to balance
```
Comment (SebastianMarian) [inline data/state/account.go]: subtracts

File: data/state/errors.go [hunk 5]
```diff
@@ -146,6 +146,9 @@ var ErrOperationNotPermitted = errors.New("operation in account not permitted")
 // ErrInvalidAddressLength signals that address length is invalid
 var ErrInvalidAddressLength = errors.New("invalid address length")
 
+// ErrOnlyPositiveValueAccepted signals that balance change operation was called with negative value
+var ErrOnlyPositiveValueAccepted = errors.New("insufficient funds")
```
Comment (SebastianMarian) [inline data/state/errors.go]: insufficient funds ?

File: process/smartContract/process.go [hunk 6]
```diff
@@ -928,6 +924,7 @@ func (sc *scProcessor) processSCOutputAccounts(
 		}
 
 		sumOfAllDiff = sumOfAllDiff.Add(sumOfAllDiff, outAcc.BalanceDelta)
```
Comment (raduchis) [inline process/smartContract/process.go]: this looks a bit odd

Comment (sasurobert) [reply process/smartContract/process.go]: refactored.

File: data/state/account.go (hunk 4 above)
Comment (sasurobert) [reply data/state/account.go]: done

File: data/state/errors.go (hunk 5 above)
Comment (sasurobert) [reply data/state/errors.go]: deleted

File: data/state/account.go (hunk 4 above)
Comment (sasurobert) [reply data/state/account.go]: done

File: data/state/errors.go (hunk 5 above)
Comment (sasurobert) [reply data/state/errors.go]: deleted
